### ISSUE TITLE
Setup Dagger 2 should explain bit more about annotation processing

### ISSUE THREAD

**Issue by Goran1708** [codepath/android_guides#281]
https://github.com/codepath/android_guides/wiki/Dependency-Injection-with-Dagger-2

**Setup** part should mention after first paragraph that after gradle plugin 2.2.0 we should remove `apply plugin: 'com.neenbedankt.android-apt'` and use `annotationProcessor` or even better move that into https://github.com/codepath/android_guides/wiki/Getting-Started-with-Gradle and link that to dagger setup page.

Basically this https://stackoverflow.com/questions/37893533/android-studio-2-2-how-to-apply-dagger2-without-android-apt-plugin

**Comment by rogerhu** [comment]
Updated here:https://github.com/codepath/android_guides/wiki/Getting-Started-with-Gradle#upgrading-gradle
